File: LoginDemo/LoginDemo/ViewModel/Common/WindowManager.vb
Imports System.Collections
Imports System.Windows


''' <summary>
''' 窗口管理器
''' </summary>
Public NotInheritable Class WindowManager
    Private Sub New()
    End Sub
    Private Shared _RegisterWindow As New Hashtable()

    Public Shared Sub Register(Of T)(key As String)
        If Not _RegisterWindow.Contains(key) Then
            _RegisterWindow.Add(key, GetType(T))
        End If
    End Sub

    Public Shared Sub Register(key As String, t As Type)
        If Not _RegisterWindow.Contains(key) Then
            _RegisterWindow.Add(key, t)
        End If
    End Sub

    Public Shared Sub Remove(key As String)
        If _RegisterWindow.ContainsKey(key) Then
            _RegisterWindow.Remove(key)
        End If
    End Sub

    Public Shared Sub Show(key As String, VM As Object)
        If _RegisterWindow.ContainsKey(key) Then
            Dim win = DirectCast(Activator.CreateInstance(DirectCast(_RegisterWindow(key), Type)), Window)
            win.DataContext = VM
            win.Show()
        End If
    End Sub
End Class

File: LoginDemo/LoginDemo/ViewModel/Common/WindowManager.cs
using System;
using System.Collections;
using System.Windows;

namespace LoginDemo.ViewModel.Common
{
    /// <summary>
    /// 窗口管理器
    /// </summary>
    public static class WindowManager
    {
        private static Hashtable _RegisterWindow = new Hashtable();

        public static void Register<T>(string key)
        {
            if (!_RegisterWindow.Contains(key))
            {
                _RegisterWindow.Add(key, typeof(T));
            }
        }

        public static void Register(string key, Type t)
        {
            if (!_RegisterWindow.Contains(key))
            {
                _RegisterWindow.Add(key, t);
            }
        }

        public static void Remove(string key)
        {
            if (_RegisterWindow.ContainsKey(key))
            {
                _RegisterWindow.Remove(key);
            }
        }

        public static void Show(string key, object VM)
        {
            if (_RegisterWindow.ContainsKey(key))
            {
                var win = (Window)Activator.CreateInstance((Type)_RegisterWindow[key]);
                win.DataContext = VM;
                win.Show();
            }
        }
    }
}
